### PR TITLE
feat: allow codegen to run in browser

### PR DESCRIPTION
**Pull request by dpilch**
#### Description of changes

- Import introspection schema.
- Replace path.win32.sep with \\
- Allows codegen to be run in the browser. This is needed for studio to generate models at runtime.

#### Codegen Paramaters Changed or Added

N/A

#### Issue #, if available

N/A

#### Description of how you validated changes

* unit tests
* E2E tests
* verify in studio with tagged release

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Changes are tested on windows. Some Node functions (such as `path`) behave differently on windows.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by palpatim — DISMISSED** [review]

File: packages/amplify-codegen-e2e-tests/test-apps/graphql-generator-app/cypress/support/commands.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+/// <reference types="cypress" />
```
Comment (sundersc) [inline packages/amplify-codegen-e2e-tests/test-apps/graphql-generator-app/cypress/support/commands.ts:1]: All the lines on this file are comments. Should we remove this file?

File: packages/amplify-codegen/src/utils/getRelativeTypesPath.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ function getRelativeTypesPath(opsGenDirectory, generatedFileName) {
     const relativePath = path
       .relative(opsGenDirectory, generatedFileName)
       // ensure posix path separators are used
-      .split(path.win32.sep)
+      .split(path.win32?.sep || '\\')
```
Comment (sundersc) [inline packages/amplify-codegen/src/utils/getRelativeTypesPath.js:8]: Why do we fallback to `\`? Does this code get executed only in windows?

Comment (dpilch) [reply packages/amplify-codegen/src/utils/getRelativeTypesPath.js:8]: This is executed in all environments. `path.win32` is not implemented by `path-browserify`. When in browser `\` will be used, when in node `path.win32.sep` (which is `\`) will be used.

File: packages/amplify-codegen-e2e-tests/test-apps/graphql-generator-app/cypress/support/commands.ts (hunk 1 above)
Comment (dpilch) [reply packages/amplify-codegen-e2e-tests/test-apps/graphql-generator-app/cypress/support/commands.ts:1]: This file was auto-generated by cypress. It needs to be present for the cypress to function. I think we could remove all of the lines though.

**Review by palpatim — APPROVED** [review]

**Review by palpatim — APPROVED** [review]

**Review by sundersc — APPROVED** [review]

**Review by phani-srikar — APPROVED** [review]

**Pull request merged**